src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx

```diff
@@ -0,0 +1,195 @@
+import Button from 'app/components/button';
+import DateTime from 'app/components/dateTime';
+import Version from 'app/components/version';
+import {RELEASES_TOUR_STEPS} from 'app/views/releases/list/releaseLanding';
+import FeatureTourModal from 'app/components/modals/featureTourModal';
+
+const DOCS_URL = 'https://docs.sentry.io/product/releases/';
+
+type Props = AsyncComponent['props'] & {
+  organization: Organization;
+  projectSlug: string;
+  projectId: string | undefined;
+  location: Location;
+};
+
+type State = {
+  releases: Release[] | null;
+  hasOlderReleases: boolean;
+} & AsyncComponent['state'];
+
+class ProjectLatestReleases extends AsyncComponent<Props, State> {
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {location, organization, projectSlug} = this.props;
+
+    const query = {
+      ...pick(location.query, [...Object.values(URL_PARAM)]),
+      per_page: 5,
+    };
+
+    // TODO(project-detail): this does not filter releases for the given time
+    return [
+      ['releases', `/projects/${organization.slug}/${projectSlug}/releases/`, {query}],
+    ];
+  }
+
+  /**
+   * If our releases are empty, determine if we had a release in the last 90 days (empty message differs then)
+   */
+  async onLoadAllEndpointsSuccess() {
+    const {releases} = this.state;
+    const {organization, projectSlug} = this.props;
+
+    if ((releases ?? []).length !== 0) {
+      this.setState({hasOlderReleases: true});
+      return;
+    }
+
+    this.setState({loading: true});
+
+    const oldReleases = await this.api.requestPromise(
+      `/projects/${organization.slug}/${projectSlug}/releases/`,
+      {
+        method: 'GET',
+        query: {
+          statsPeriod: '90d',
+          per_page: 1,
+        },
+      }
+    );
+
+    this.setState({hasOlderReleases: oldReleases.length > 0, loading: false});
+  }
+
+  handleTourAdvance = (index: number) => {
+    const {organization, projectId} = this.props;
+
+    analytics('releases.landing_card_clicked', {
+      org_id: parseInt(organization.id, 10),
+      project_id: projectId && parseInt(projectId, 10),
+      step_id: index,
+      step_title: RELEASES_TOUR_STEPS[index].title,
+    });
+  };
+
+  renderReleaseRow = (release: Release) => {
+    const {projectId} = this.props;
+    const {lastDeploy, dateCreated} = release;
+
+    return (
+      <React.Fragment>
+        <DateTime date={lastDeploy?.dateFinished || dateCreated} seconds={false} />
+        <VersionWrapper>
+          <Version
+            version={release.version}
+            tooltipRawVersion
+            projectId={projectId}
+            truncate
+          />
+        </VersionWrapper>
+      </React.Fragment>
+    );
+  };
+
+  renderInnerBody() {
+    const {loading, releases, hasOlderReleases} = this.state;
+    const checkingForOlderReleases =
+      (releases ?? []).length === 0 && hasOlderReleases === undefined ? true : false;
```

Comment:
can `hasOlderReleases` be `undefined`? in the state is only of type boolean. should we add a `getDefaultState` function so then it will be always boolean?

Reply:
We want it to be `undefined` at first, I marked it as optional in state type.